File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/Utils/LocaleTranslatorUtils.java
package com.lattis.ellipse.Utils;

import android.content.Context;

public class LocaleTranslatorUtils {

    public static String getLocaleString(Context context, String word){
        return  getStringResourceByName(context,word);
    }

    private static String getStringResourceByName(Context context,String word) {
        try {
            if (word != null) {
                String packageName = context.getPackageName();
                int resId = context.getResources().getIdentifier(word.toLowerCase(), "string", packageName);
                return context.getString(resId);
            }
        }catch(Exception r){

        }
        return word;
    }

}

File: velolabs_repos/lattis_android_v2/app/src/main/java/com/lattis/lattis/presentation/utils/LocaleTranslatorUtils.kt
package com.lattis.lattis.presentation.utils

import android.content.Context
import com.lattis.lattis.utils.UtilsHelper.isNumber

object LocaleTranslatorUtils {
    fun getLocaleString(context: Context, word: String?,count:String?=null): String? {
        return if(count!=null && isNumber(count) && count.toIntOrNull()!=null){
            getQuantityResourceByName(context,word,count.toInt())
        }else {
            getStringResourceByName(context,word)
        }
    }

    private fun getQuantityResourceByName(
                                          context: Context,
                                          word: String?,
                                          count:Int
    ): String? {
        try {
            if (word != null) {
                val packageName = context.packageName
                val resId =
                    context.resources.getIdentifier(word.lowercase(), "plurals", packageName)
                return context.resources.getQuantityString(resId,count,count)
            }
        } catch (r: Exception) {

        }
        return getStringResourceByName(context,word)
    }

    private fun getStringResourceByName(
        context: Context,
        word: String?
    ): String? {
        try {
            if (word != null) {
                val packageName = context.packageName
                val resId =
                    context.resources.getIdentifier(word.lowercase(), "string", packageName)
                return context.getString(resId)
            }
        } catch (r: Exception) {
        }
        return word
    }
}